src/waitForElementToBeRemoved.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+import waitFor from './waitFor';
+import type { WaitForOptions } from './waitFor';
+
+const isRemoved = (result) =>
+  !result || (Array.isArray(result) && !result.length);
+
+export default async function waitForElementToBeRemoved<T>(
+  expectation: () => T,
+  options?: WaitForOptions
+): Promise<null> {
```

Comment:
Not sure if `null` is a proper type here. But we do not have anything meaningful to return.
RTL declares its version in TS as `Promise<T>` but uses `return true` in their impl.

Reply:
IMO I would keep the `Promise<T>` return type and return the initial result of `expectation` call. Since in code we require that initially `expectation` is successful (returns something) and only then starts throwing/returning nothing.

Reply:
yup, let's do it

Reply:
Done

Reply:
Also logged issue in Dom TL that they have similar discrepancy.